### PR TITLE
feat: basic cart setup

### PR DESCRIPTION
**Pull request by VincentRPS**

### CODE REVIEW

File: crates/cart/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// SPDX-License-Identifier: MIT
+
+#![feature(let_chains)]
```
Comment (elenakrittik) [inline crates/cart/src/main.rs:3]: I believe this is not needed

File: crates/cart/src/config.rs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+
+use crate::error::Error;
+
+pub trait VerifyConfig {
+    fn verify_config(&self) -> Result<(), Error>;
+}
```
[the remark this answers is not in the document]
Comment (VincentRPS) [reply crates/cart/src/config.rs:11]: 1. to keep uniformity among the config structs and 2. to allow people to use cart as a library and use different types of configs

[the remark this answers is not in the document]
Comment (elenakrittik) [reply crates/cart/src/config.rs:11]: I really doubt that custom configs would be a particularly useful feature, but the first point is enough already

[the remark this answers is not in the document]
Comment (elenakrittik) [reply crates/cart/src/config.rs:11]: Especially because the trait hard-codes the error variant, which cannot be extended

**Review by elenakrittik — APPROVED** [review]

**Pull request merged**